src/org/labkey/test/tests/component/HeatMapTest.java

```diff
@@ -0,0 +1,157 @@
+            "     SUM(CASE WHEN (age_in_months(M.Created, curdate()) < 12) THEN 1 ELSE 0 END)\n" +
+            "                    AS InRangeCount\n" +
+            "   FROM Materials AS M\n" +
+            "   GROUP BY SampleSet\n" +
+            "  ) AS TOTAL\n" +
+            "\n" +
+            "  FULL JOIN\n" +
+            "\n" +
+            "  (SELECT\n" +
+            "     M.SampleSet                                AS _PivotSampleSet @hidden,\n" +
+            "     COUNT(M.RowID)                             AS MonthCount,\n" +
+            "     CAST(YEAR(M.Created) AS VARCHAR) || '-' || CAST(MONTH(M.Created) AS VARCHAR) AS YearMonth\n" +
+            "   FROM exp.Materials AS M\n" +
+            "   WHERE (age_in_months(M.Modified, curdate()) < 12)\n" +
+            "   GROUP BY M.SampleSet,\n" +
+            "     CAST(YEAR(M.Created) AS VARCHAR) || '-' || CAST(MONTH(M.Created) AS VARCHAR)\n" +
+            "   PIVOT MonthCount BY YearMonth\n" +
+            "  ) AS P\n" +
+            "    ON P._PivotSampleSet = TOTAL.SampleSet\n" +
+            "\n" +
+            "WHERE SampleSet.Name NOT IN ('RawMaterials', 'MixtureBatches')";
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject() throws Exception
+    {
+        HeatMapTest init = (HeatMapTest) getCurrentTest();
+
+        init.doSetup();
+    }
+
+    private void doSetup() throws Exception
+    {
+        _containerHelper.createProject(getProjectName(), null);
+        createQuery(getProjectName(), "SampleSetHeatMap", "exp", HEATMAP_QUERY, null, false);
+
+        TEST_SAMPLE_TYPES.put("basic_heatmap_samples 1", 50);
+        TEST_SAMPLE_TYPES.put("more_basic_samples", 30);
+        TEST_SAMPLE_TYPES.put("filler_samples", 75);
+        TEST_SAMPLE_TYPES.put("heatmap_samples", 100);
+
+        for(String sampleTypeName : TEST_SAMPLE_TYPES.keySet())
+        {
+            makeHeatmapSampletype(sampleTypeName, TEST_SAMPLE_TYPES.get(sampleTypeName));
+        }
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+    }
+
+    @Test
+    public void testHeatMap()
+    {
+        CoreComponentsTestPage testPage = CoreComponentsTestPage.beginAt(this, getProjectName());
+        HeatMap heatMap = testPage.getHeatMap();
+
+        assertTrue("Heatmap should be loaded and have data", heatMap.hasData());
+        for(String sampleTypeName : TEST_SAMPLE_TYPES.keySet())
+        {
+            String expectedSampleCount = Integer.toString(TEST_SAMPLE_TYPES.get(sampleTypeName));
+            assertThat("expect ["+expectedSampleCount+"] samples",
+                    heatMap.getSummaryLink(sampleTypeName).getText(), is(expectedSampleCount));
+        }
+        assertThat("", heatMap.getEntityNames(), hasItems(TEST_SAMPLE_TYPES.keySet().toArray(new String[0])));
```

Comment:
Add non-blank message.
```suggestion
        assertThat("something", heatMap.getEntityNames(), hasItems(TEST_SAMPLE_TYPES.keySet().toArray(new String[0])));
```